go/vt/tabletserver/query_executor.go

```diff
@@ -325,6 +335,68 @@ func (qre *QueryExecutor) execPKIN() (*sqltypes.Result, error) {
 	return qre.fetchMulti(pkRows, limit)
 }
 
+func (qre *QueryExecutor) execNextval() (*sqltypes.Result, error) {
+	t := qre.plan.TableInfo
+	t.Seq.Lock()
+	defer t.Seq.Unlock()
+	if t.NextVal >= t.LastVal {
+		_, err := qre.execAsTransaction(func(conn *TxConnection) (*sqltypes.Result, error) {
+			query := fmt.Sprintf("select next_id, chunk_size, increment from `%s` where id = 0", qre.plan.TableName)
+			conn.RecordQuery(query)
+			qr, err := qre.execSQL(conn, query, false)
+			if err != nil {
+				return nil, err
+			}
+			if len(qr.Rows) != 1 {
+				return nil, fmt.Errorf("unexpected rows from reading sequence %s: %d", qre.plan.TableName, len(qr.Rows))
+			}
+			nextID, err := qr.Rows[0][0].ParseInt64()
+			if err != nil {
+				return nil, fmt.Errorf("error loading sequence %s: %v", qre.plan.TableName, err)
+			}
+			chunkSize, err := qr.Rows[0][1].ParseInt64()
```

Comment:
chunkSize is actually cache size, which is the number of values we cache.

So you need to increment next_id by 'cache * increment', not just increment.